pkg/gui/context/sub_commits_context.go

```diff
@@ -71,12 +73,45 @@ func NewSubCommitsContext(
 			selectedCommitSha,
 			startIdx,
 			endIdx,
-			shouldShowGraph(c),
+			// Don't show the graph in the left/right view; we'd like to, but
+			// it's too complicated:
+			shouldShowGraph(c) && viewModel.GetRefToShowDivergenceFrom() == "",
 			git_commands.NewNullBisectInfo(),
 			false,
 		)
 	}
 
+	getNonModelItems := func() []*NonModelItem {
+		result := []*NonModelItem{}
+		if viewModel.GetRefToShowDivergenceFrom() != "" {
+			_, upstreamIdx, found := lo.FindIndexOf(
+				c.Model().SubCommits, func(c *models.Commit) bool { return c.Divergence == models.DivergenceRight })
+			if !found {
+				upstreamIdx = 0
+			}
+			result = append(result, &NonModelItem{
+				Index:      upstreamIdx,
+				ClientData: fmt.Sprintf("--- %s ---", c.Tr.DivergenceSectionHeaderRemote),
```

Comment:
Happy to take suggestions for a better UI, for now I went with the simplest possible design. I wasn't sure if we want to align this with the sha column or maybe the subject column.